app/preview.py

```diff
@@ -86,19 +99,41 @@ def view_letter_template(filetype):
     if filetype == "pdf" and request.args.get("page") is not None:
         abort(400)
 
-    html = get_html(get_and_validate_json_from_request(request, preview_schema))
+    json = get_and_validate_json_from_request(request, preview_schema)
+    html = get_html(json)
+    pdf = get_pdf(html)
 
     if filetype == "pdf":
+        # TODO: handle attachments
         return send_file(
-            path_or_file=get_pdf(html),
+            path_or_file=pdf,
             mimetype="application/pdf",
         )
     elif filetype == "png":
-        return send_file(
-            path_or_file=get_png(
+        templated_letter_page_count = get_page_count(pdf)
+        requested_page = int(request.args.get("page", 1))
+
+        if requested_page <= templated_letter_page_count:
+            png_preview = get_png(
                 html,
-                int(request.args.get("page", 1)),
-            ),
+                requested_page,
+            )
+        elif letter_attachment := json["template"].get("letter_attachment"):
+            # get attachment page instead
+            requested_attachment_page = requested_page - templated_letter_page_count
+
+            attachment_pdf = get_attachment_pdf(letter_attachment["url"])
+            encoded_string = base64.b64encode(attachment_pdf)
+            png_preview = get_png_from_precompiled(
+                encoded_string=encoded_string,
+                page_number=requested_attachment_page,
+                hide_notify=False,
+            )
+        else:
+            abort(400, f"Letter does not have a page {requested_page}")
```

Comment:
I think we have a logical error here. If we have a template with 5 pages, and ask for page 6, we'll get this error. If we have a template with 4 pages and 2 pages of attachment, and ask for page 7, we won't throw this error.